diskio_plugin does not handle empty args in msg_curse()

The msg_curse function is defined as follows, with the important line being 230:
https://github.com/nicolargo/glances/blob/6bbc78da6fa3170b35b11e828ed8e76709855a7e/glances/plugins/diskio/__init__.py#L210-L235

but the diskio_plugin test calls it like so (see line 265 for example):
https://github.com/nicolargo/glances/blob/6bbc78da6fa3170b35b11e828ed8e76709855a7e/tests/test_plugin_diskio.py#L259-L278


This leads to the test error when building:
<details>
```log
        # Header
        msg = '{:{width}}'.format('DISK I/O', width=name_max_width)
        ret.append(self.curse_add_line(msg, "TITLE"))
>       if args.diskio_iops:
           ^^^^^^^^^^^^^^^^
E       AttributeError: 'NoneType' object has no attribute 'diskio_iops'

glances/plugins/diskio/__init__.py:230: AttributeError
____________ TestDiskioPluginMsgCurse.test_msg_curse_with_max_width ____________

self = <test_plugin_diskio.TestDiskioPluginMsgCurse object at 0x7ffff426fa80>
diskio_plugin = {'nvme0n1': {'read_count': 163, 'write_count': 30, 'read_bytes': 3342336, 'write_bytes': 311296, 'read_time': >

    def test_msg_curse_with_max_width(self, diskio_plugin):
        """Test that msg_curse works with max_width."""
        diskio_plugin.update()
>       msg = diskio_plugin.msg_curse(max_width=80)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/test_plugin_diskio.py:277:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = {'nvme0n1': {'read_count': 163, 'write_count': 30, 'read_bytes': 3342336, 'write_bytes': 311296, 'read_time': 695, 'wr.>
args = None, max_width = 80


        # Only process if stats exist and display plugin enable...
        if not self.stats or self.is_disabled():

        # Only process if stats exist and display plugin enable...
        if not self.stats or self.is_disabled():

        # Only process if stats exist and display plugin enable...
        if not self.stats or self.is_disabled():

        # Only process if stats exist and display plugin enable...
        if not self.stats or self.is_disabled():

        # Only process if stats exist and display plugin enable...
        if not self.stats or self.is_disabled():

        # Only process if stats exist and display plugin enable...
        if not self.stats or self.is_disabled():
            return ret

        # Only process if stats exist and display plugin enable...
        if not self.stats or self.is_disabled():
            return ret

        # Max size for the interface name
        if max_width:
            name_max_width = max_width - 13
        else:
            # No max_width defined, return an empty curse message
            logger.debug(f"No max_width defined for the {self.plugin_name} plugin, it will not be displayed.")
            return ret

        # Header
        msg = '{:{width}}'.format('DISK I/O', width=name_max_width)
        ret.append(self.curse_add_line(msg, "TITLE"))
>       if args.diskio_iops:
           ^^^^^^^^^^^^^^^^
E       AttributeError: 'NoneType' object has no attribute 'diskio_iops'

glances/plugins/diskio/__init__.py:230: AttributeError
```
</details>

--- a/glances/plugins/containers/__init__.py
+++ b/glances/plugins/containers/__init__.py
@@ -471,7 +471,7 @@ class ContainersPlugin(GlancesPluginModel):
 
     def build_net_line(self, args):
         def build_with_this_args(ret, container):
-            if args.byte:
+            if args and args.byte:
                 # Bytes per second (for dummy)
                 to_bit = 1
                 unit = ''

--- a/glances/plugins/diskio/__init__.py
+++ b/glances/plugins/diskio/__init__.py
@@ -227,12 +227,12 @@ class DiskioPlugin(GlancesPluginModel):
         # Header
         msg = '{:{width}}'.format('DISK I/O', width=name_max_width)
         ret.append(self.curse_add_line(msg, "TITLE"))
-        if args.diskio_iops:
+        if args and args.diskio_iops:
             msg = '{:>8}'.format('IOR/s')
             ret.append(self.curse_add_line(msg))
             msg = '{:>7}'.format('IOW/s')
             ret.append(self.curse_add_line(msg))
-        elif args.diskio_latency:
+        elif args and args.diskio_latency:
             msg = '{:>8}'.format('ms/opR')
             ret.append(self.curse_add_line(msg))
             msg = '{:>7}'.format('ms/opW')
@@ -256,7 +256,7 @@ class DiskioPlugin(GlancesPluginModel):
                 disk_name = disk_name[:name_max_width] + '_'
             msg = '{:{width}}'.format(nativestr(disk_name), width=name_max_width + 1)
             ret.append(self.curse_add_line(msg))
-            if args.diskio_iops:
+            if args and args.diskio_iops:
                 # count
                 txps = self.auto_unit(i.get('read_count_rate_per_sec', None))
                 rxps = self.auto_unit(i.get('write_count_rate_per_sec', None))
@@ -272,7 +272,7 @@ class DiskioPlugin(GlancesPluginModel):
                         msg, self.get_views(item=i[self.get_key()], key='write_count', option='decoration')
                     )
                 )
-            elif args.diskio_latency:
+            elif args and args.diskio_latency:
                 # latency (mean time spent reading/writing per operation)
                 txps = self.auto_unit(i.get('read_latency', None), low_precision=True)
                 rxps = self.auto_unit(i.get('write_latency', None), low_precision=True)

--- a/glances/plugins/fs/__init__.py
+++ b/glances/plugins/fs/__init__.py
@@ -294,7 +294,7 @@ class FsPlugin(GlancesPluginModel):
         # Header
         msg = '{:{width}}'.format('FILE SYS', width=name_max_width)
         ret.append(self.curse_add_line(msg, "TITLE"))
-        if args.fs_free_space:
+        if args and args.fs_free_space:
             msg = '{:>8}'.format('Free')
         else:
             msg = '{:>8}'.format('Used')
@@ -314,7 +314,7 @@ class FsPlugin(GlancesPluginModel):
                 mnt_point = mnt_point[:name_max_width] + '_'
             msg = '{:{width}}'.format(nativestr(mnt_point), width=name_max_width + 1)
             ret.append(self.curse_add_line(msg))
-            if args.fs_free_space:
+            if args and args.fs_free_space:
                 msg = '{:>7}'.format(self.auto_unit(i['free']))
             else:
                 msg = '{:>7}'.format(self.auto_unit(i['used']))

--- a/glances/plugins/gpu/__init__.py
+++ b/glances/plugins/gpu/__init__.py
@@ -268,9 +268,9 @@ class GpuPlugin(GlancesPluginModel):
         ret.append(self.curse_new_line())
         ret.append(self.curse_add_line('{:13}'.format('temp mean:' if is_multi else 'temperature:')))
         mean_temp = self._get_mean('temperature')
-        if mean_temp is not None and args.fahrenheit:
+        if mean_temp is not None and args and args.fahrenheit:
             mean_temp = to_fahrenheit(mean_temp)
-        unit = 'F' if args.fahrenheit else 'C'
+        unit = 'F' if args and args.fahrenheit else 'C'
         ret.append(
             self.curse_add_line(
                 self._format_value(mean_temp, unit),

--- a/glances/plugins/load/__init__.py
+++ b/glances/plugins/load/__init__.py
@@ -164,7 +164,7 @@ class LoadPlugin(GlancesPluginModel):
             ret.append(self.curse_new_line())
             msg = '{:7}'.format(f'{load_time} min')
             ret.append(self.curse_add_line(msg))
-            if args.disable_irix and log_core() != 0:
+            if args and args.disable_irix and log_core() != 0:
                 # Enable Irix mode for load (see issue #1554)
                 load_stat = self.stats[f'min{load_time}'] / log_core() * 100
                 msg = f'{load_stat:>5.1f}%'

--- a/glances/plugins/network/__init__.py
+++ b/glances/plugins/network/__init__.py
@@ -255,9 +255,9 @@ class NetworkPlugin(GlancesPluginModel):
         # Header
         msg = '{:{width}}'.format('NETWORK', width=name_max_width)
         ret.append(self.curse_add_line(msg, "TITLE"))
-        if args.network_cumul:
+        if args and args.network_cumul:
             # Cumulative stats
-            if args.network_sum:
+            if args and args.network_sum:
                 # Sum stats
                 msg = '{:>14}'.format('Rx+Tx')
                 ret.append(self.curse_add_line(msg))
@@ -269,7 +269,7 @@ class NetworkPlugin(GlancesPluginModel):
                 ret.append(self.curse_add_line(msg))
         else:
             # Bitrate stats
-            if args.network_sum:
+            if args and args.network_sum:
                 # Sum stats
                 msg = '{:>14}'.format('Rx+Tx/s')
                 ret.append(self.curse_add_line(msg))
@@ -296,7 +296,7 @@ class NetworkPlugin(GlancesPluginModel):
                 # Cut interface name if it is too long
                 if_name = '_' + if_name[-name_max_width + 1 :]
 
-            if args.byte:
+            if args and args.byte:
                 # Bytes per second (for dummy)
                 to_bit = 1
                 unit = ''
@@ -305,7 +305,7 @@ class NetworkPlugin(GlancesPluginModel):
                 to_bit = 8
                 unit = 'b'
 
-            if args.network_cumul and i.get('bytes_recv') is not None and i.get('bytes_sent') is not None:
+            if args and args.network_cumul and i.get('bytes_recv') is not None and i.get('bytes_sent') is not None:
                 rx = self.auto_unit(int(i['bytes_recv'] * to_bit)) + unit
                 tx = self.auto_unit(int(i['bytes_sent'] * to_bit)) + unit
                 ax = self.auto_unit(int(i['bytes_all'] * to_bit)) + unit
@@ -322,7 +322,7 @@ class NetworkPlugin(GlancesPluginModel):
             ret.append(self.curse_new_line())
             msg = '{:{width}}'.format(if_name, width=name_max_width)
             ret.append(self.curse_add_line(msg))
-            if args.network_sum:
+            if args and args.network_sum:
                 msg = f'{ax:>14}'
                 ret.append(self.curse_add_line(msg))
             else:

--- a/glances/plugins/processcount/__init__.py
+++ b/glances/plugins/processcount/__init__.py
@@ -100,7 +100,7 @@ class ProcesscountPlugin(GlancesPluginModel):
         ret = []
 
         # Only process if stats exist and display plugin enable...
-        if args.disable_process:
+        if args and args.disable_process:
             msg = "PROCESSES DISABLED (press 'z' to display)"
             ret.append(self.curse_add_line(msg))
             return ret

--- a/glances/plugins/processlist/__init__.py
+++ b/glances/plugins/processlist/__init__.py
@@ -220,7 +220,7 @@ class ProcesslistPlugin(GlancesPluginModel):
             glances_processes.set_sort_key(config.as_dict()['processlist']['sort_key'], False)
         if 'export' in config.as_dict()['processlist']:
             glances_processes.export_process_filter = config.as_dict()['processlist']['export']
-            if args.export:
+            if args and args.export:
                 logger.info("Export process filter is set to: {}".format(config.as_dict()['processlist']['export']))
         if 'focus' in config.as_dict()['processlist']:
             glances_processes.process_focus = config.as_dict()['processlist']['focus']
@@ -302,7 +302,7 @@ class ProcesslistPlugin(GlancesPluginModel):
         """Return process CPU curses"""
         if key_exist_value_not_none_not_v('cpu_percent', p, ''):
             cpu_layout = self.layout_stat['cpu'] if p['cpu_percent'] < 100 else self.layout_stat['cpu_no_digit']
-            if args.disable_irix and self.nb_log_core != 0:
+            if args and args.disable_irix and self.nb_log_core != 0:
                 msg = cpu_layout.format(p['cpu_percent'] / float(self.nb_log_core))
             else:
                 msg = cpu_layout.format(p['cpu_percent'])
@@ -505,7 +505,7 @@ class ProcesslistPlugin(GlancesPluginModel):
                     msg = self.layout_stat['command'].format(cmd)
                     ret.append(self.curse_add_line(msg, decoration=process_decoration, splittable=True))
                 if arguments:
-                    msg = ' ' if args.cursor_process_name_position == 0 else unicode_message('THREE_DOTS')
+                    msg = ' ' if args and args.cursor_process_name_position == 0 else unicode_message('THREE_DOTS')
                     msg += self.layout_stat['command'].format(arguments[args.cursor_process_name_position :])
                     ret.append(self.curse_add_line(msg, splittable=True))
             else:
@@ -588,7 +588,7 @@ class ProcesslistPlugin(GlancesPluginModel):
 
         # A filter is set Display the stats summaries
         if glances_processes.process_filter is not None:
-            if args.reset_minmax_tag:
+            if args and args.reset_minmax_tag:
                 args.reset_minmax_tag = not args.reset_minmax_tag
                 self._mmm_reset()
             self._msg_curse_sum(ret, args=args)
@@ -756,9 +756,9 @@ class ProcesslistPlugin(GlancesPluginModel):
     def _msg_curse_header_cpu(self, ret, process_sort_key, display_stats, sort_style='SORT', args=None):
         """Build the header and add it to the ret dict."""
         if 'cpu_percent' in display_stats:
-            if args.disable_irix and 0 < self.nb_log_core < 10:
+            if args and args.disable_irix and 0 < self.nb_log_core < 10:
                 msg = self.layout_header['cpu'].format('CPU%/' + str(self.nb_log_core))
-            elif args.disable_irix and self.nb_log_core != 0:
+            elif args and args.disable_irix and self.nb_log_core != 0:
                 msg = self.layout_header['cpu'].format('CPUi')
             else:
                 msg = self.layout_header['cpu'].format('CPU%')
@@ -840,7 +840,7 @@ class ProcesslistPlugin(GlancesPluginModel):
         if 'cpu_num' in display_stats:
             msg = self.layout_header['processor'].format('CPU')
             ret.append(self.curse_add_line(msg, optional=True))
-        if args.is_standalone and not args.disable_cursor:
+        if args and args.is_standalone and not args.disable_cursor:
             shortkey = "('e' to pin | 'k' to kill)"
         else:
             shortkey = ""

--- a/glances/plugins/system/__init__.py
+++ b/glances/plugins/system/__init__.py
@@ -235,7 +235,7 @@ class SystemPlugin(GlancesPluginModel):
             return ret
 
         # Build the string message
-        if args.client:
+        if args and args.client:
             # Client mode
             if args.cs_status.lower() == "connected":
                 msg = 'Connected to '

--- a/tests/test_plugin_diskio.py
+++ b/tests/test_plugin_diskio.py
@@ -262,18 +262,21 @@ class TestDiskioPluginMsgCurse:
     def test_msg_curse_returns_list(self, diskio_plugin):
         """Test that msg_curse returns a list."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         msg = diskio_plugin.msg_curse(max_width=80)
         assert isinstance(msg, list)
 
     def test_msg_curse_empty_without_max_width(self, diskio_plugin):
         """Test that msg_curse returns empty without max_width."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         msg = diskio_plugin.msg_curse()
         assert isinstance(msg, list)
 
     def test_msg_curse_with_max_width(self, diskio_plugin):
         """Test that msg_curse works with max_width."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         msg = diskio_plugin.msg_curse(max_width=80)
         assert isinstance(msg, list)
 
@@ -284,12 +287,14 @@ class TestDiskioPluginSorting:
     def test_sorted_stats_returns_list(self, diskio_plugin):
         """Test that sorted_stats returns a list."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         sorted_stats = diskio_plugin.sorted_stats()
         assert isinstance(sorted_stats, list)
 
     def test_sorted_stats_preserves_count(self, diskio_plugin):
         """Test that sorted_stats preserves disk count."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         raw_count = len(diskio_plugin.get_raw())
         sorted_count = len(diskio_plugin.sorted_stats())
         assert raw_count == sorted_count
@@ -301,12 +306,14 @@ class TestDiskioPluginExport:
     def test_get_export_returns_list(self, diskio_plugin):
         """Test that get_export returns a list."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         export = diskio_plugin.get_export()
         assert isinstance(export, list)
 
     def test_export_equals_raw(self, diskio_plugin):
         """Test that export equals raw stats by default."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         assert diskio_plugin.get_export() == diskio_plugin.get_raw()
 
 
@@ -316,6 +323,7 @@ class TestDiskioPluginAlerts:
     def test_views_have_decoration(self, diskio_plugin):
         """Test that disk views have decoration after update."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         time.sleep(0.1)
         diskio_plugin.update()
         diskio_plugin.update_views()
@@ -334,6 +342,7 @@ class TestDiskioPluginAlias:
     def test_alias_field_may_be_present(self, diskio_plugin):
         """Test that alias field may be present for disks."""
         diskio_plugin.update()
+        diskio_plugin.update_views()
         stats = diskio_plugin.get_raw()
         # Alias is optional, so just verify the field handling works
         for disk in stats:

